animal-sniffer/src/main/java/org/codehaus/mojo/animal_sniffer/SignatureChecker.java

```diff
@@ -178,6 +188,46 @@ protected void process(final String name, InputStream image) throws IOException
         }
     }
 
+    /**
+     * First-pass visitor to collect Kotlin $annotations methods that have @IgnoreJRERequirement.
+     * Kotlin places property annotations on synthetic $annotations methods rather than the actual getter/setter.
+     */
+    private class AnnotationCollector extends ClassVisitor {
+        private final Set<String> kotlinAnnotatedMethods = new HashSet<>();
+
+        public AnnotationCollector() {
+            super(Opcodes.ASM9);
+        }
+
+        public Set<String> getKotlinAnnotatedMethods() {
+            return kotlinAnnotatedMethods;
+        }
+
+        @Override
+        public MethodVisitor visitMethod(int access, String name, String desc, String signature, String[] exceptions) {
+            // Only process synthetic $annotations methods (check synthetic flag first for performance)
+            if ((access & Opcodes.ACC_SYNTHETIC) != 0 && name.endsWith(KOTLIN_ANNOTATIONS_SUFFIX)) {
+                return new MethodVisitor(Opcodes.ASM9) {
+                    @Override
+                    public AnnotationVisitor visitAnnotation(String annoDesc, boolean visible) {
+                        // Check if this is an ignore annotation
+                        for (String ignoredAnnoDesc : annotationDescriptors) {
+                            if (annoDesc.equals(ignoredAnnoDesc)) {
```

Comment:
There is already `isIgnoreAnnotation(String)` in this method; maybe it should be used instead of manually looping over the descriptors?